docs/rst/tutorials/micro/custom_platforms/custom_platforms.rst

```diff
@@ -0,0 +1,341 @@
+.. _tutorials_micro_custom_platforms:
+
+Integrating micro-ROS
+=====================
+
+Integrating micro-ROS in a platform is a process that highly depends on the target platform.
+In general, micro-ROS provides ready-to-use solutions for integrating the micro-ROS Client library in multiple platforms and build systems, more information about this micro-ROS modules can be found in :ref:`Build System Components <micro_build_system_components>` section.
+
+This tutorial aims to go beyond those modules and provide some ideas on how to integrate micro-ROS in a new platform.
+This task can be divided in two main parts: generating the micro-ROS Client library and header directory and linking it against the target embedded application.
+
+It is important to note that micro-ROS Client library is designed to be platform independent.
+This means that the library can be built with independently from the libraries of the target platform, but with the same toolchain and an ``libc`` implementation.
+
+This tutorial will cover the former topics along these sections:
+
+.. contents:: Table of Contents
+    :depth: 3
+    :local:
+    :backlinks: none
+
+
+Generating a micro-ROS Client library
+-------------------------------------
+
+The micro-ROS Client library, in most cases, is compound of:
+
+- A static library built with an specific toolchain. Normally a ``.a`` file.
+- A include folder where all the required headers are located. Normally a ``include`` folder.
+
+Given that, most of common build system tools such as CMake or Make will be able to link against the static library and use the include folder to compile the application.
+
+So in order to generate those two components, two approaches are provided: using a micro-ROS tool for generating them or creating a custom script for handling this build.
+
+micro-ROS generate_lib script
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. note::
+
+    This sections starts from a previously created micro-ROS environment. Check the first steps of `Getting started micro-ROS <https://docs.vulcanexus.org/en/latest/rst/tutorials/micro/getting_started/getting_started.html>`_ for instructions on how to create a micro-ROS environment for embedded platforms.
+
+The ``micro_ros_setup`` tool provides a script for generating and building the micro-ROS Client library according to a specific configuration and build parameters.
+
+The following command will download all the required packages.
+
+.. code-block:: bash
+
+    ros2 run micro_ros_setup create_firmware_ws.sh generate_lib
+
+For configuring both the micro-ROS library and the build process a **colcon.meta** file and a **CMake toolchain** are required.
+
+An example of a **colcon.meta** file can be:
```

Comment:
Maybe we can leave memory elements (nodes, ...) out and explain that this is the minimum required configuration to build as a static library on a custom platform.